### PR TITLE
fix(scale): return error message in proper format

### LINKED ISSUE #831: Array response when trying to scale a process to less than 0
`POST /v2/apps/<id>/scale/` Body: `{"foo": -1}`

will return 400: `["Must be greater than or equal to zero"]`

This error is in a different format from every other API error I know of.

I would expect the error to be: `{"foo":["Must be greater than or equal to zero"]}` or  `{"details":["Must be greater than or equal to zero"]}`

### PR DESCRIPTION
**Pull request by kmala**
# Summary of Changes

Handle the case of validation error instead of value error as it value error re-thrown as validation error in the validate function.

# Issue(s) that this PR Closes

fixes #831

# Associated [End To End Test](https://github.com/deis/workflow-e2e) PR(s)

# Associated [Documentation](https://github.com/deis/workflow) PR(s)

# Associated [Design Document](http://docs.deis.io/en/latest/contributing/design-documents)(s)

# Testing Instructions

Please provide a detailed list for how to test the changes in this PR.

1. Create a Deis Cluster
2. Register an app
3. Try to call the scale endpoint as mentioned in the #831 and you should get a proper json error response.

# Pull Request Hygiene TODOs

Please make sure the below checklist is complete.

- [x] Your pull request is concise and to the point (make another PR for refactoring nearby code)
- [x] Your commits are squashed into logical units of work
- [x] Your commits follow the [commit style guidelines](http://docs.deis.io/en/latest/contributing/standards/#commit-style-guide)

:cherry_blossom: Thank you! :cherry_blossom:

### CODE REVIEW

**Comment by bacongobbler** [comment]
can we add some unit tests here to validate this?

File: rootfs/api/tests/test_pods.py [hunk 1]
```diff
@@ -415,6 +415,8 @@ def test_scale_errors(self, mock_requests):
         body = {'web': -1}
         response = self.client.post(url, body)
         self.assertEqual(response.status_code, 400, response.data)
+        self.assertEqual(response.data, {'detail': "Invalid scaling format: "
+                                         "['Must be greater than or equal to zero']"})
```
Comment (Joshua-Anderson) [inline rootfs/api/tests/test_pods.py]: Is this valid JSON?

Comment (bacongobbler) [inline rootfs/api/tests/test_pods.py]: Yes, it just looks confusing. the key is `detail`, the value is a string: `Invalid scaling format: ['Must be greater than or equal to zero']`. It just looks off here,

Comment (Joshua-Anderson) [inline rootfs/api/tests/test_pods.py]: Having a value with datatypes `Invalid scaling format: ['Must be greater than or equal to zero']` is a little tricky to parse, and also doesn't really follow standard json usage. Something like this makes the error much easier to work with and is a lot less confusing for people trying to understand it.

```json
{
    "detail": {
        "scaling_format": ["Must be greater than or equal to zero"]
    }
}
```

Comment (bacongobbler) [inline rootfs/api/tests/test_pods.py]: I agree, but that's out of the scope of this issue. Mind filing an issue for it?

**Comment by helgi** [comment]
We could embed the "invalid scaling format" as part of the actual error message so that we don't have a string + list and instead just get a list. Thoughts?

**Comment by helgi** [comment]
ah wait that might be hard since we are transforming ValidationError into a DeisException...

**Pull request merged**